src/ttrss_api.cpp

```diff
@@ -403,21 +417,13 @@ void ttrss_api::fetch_feeds_per_category(
 		return;
 
 	for (json& feed : feed_list_obj) {
-		const int feed_id            = feed["id"];
-		const std::string feed_title = feed["title"];
-		const std::string feed_url   = feed["feed_url"];
-
-		std::vector<std::string> tags;
-		tags.push_back(std::string("~") + feed_title);
-
-		if (!cat_name.is_null()) {
-			tags.push_back(cat_name.get<std::string>());
-		}
-
-		auto url = strprintf::fmt("%s#%d", feed_url, feed_id);
-		feeds.push_back(tagged_feedurl(url, tags));
-
-		// TODO: cache feed_id -> feed_url (or feed_url -> feed_id ?)
+		feeds.push_back(feed_from_json(feed, [&](const json&){
+				std::vector<std::string> tags;
```

Comment:
This whole calculation doesn't depend on the feed, so you can move it out of the loop and replace this body with simple `return pre_calculated_vector`.

Reply:
Good catch! The lambda is now eliminated.

Reply:
The lambda is gone, but the calculation still happens inside the loop. The result is constant, so I don't see a reason not to move it out of the loop.

Reply:
And this is why you do code reviews :) You're absolutely right, should be fixed now.